### PR TITLE
Add an initial version of wavy bonds

### PR DESCRIPTION
**Pull request by greglandrum**
Used for attachment points and unknown stereochem
This is not a perfect implementation, but is a decent place to start
Currently only supports SVG and Cairo canvases.

Example images:
image

image

### CODE REVIEW

**Comment by bp-kelley** [comment]
Pretty.  How to the waves scale with rendering size?

**Comment by greglandrum** [comment]
These are using the SVG (or Cairo) curve primitives and the control points are scaled the same way as the bonds themselves, so they should scale appropriately.

File: Code/GraphMol/MolDraw2D/MolDraw2DCairo.cpp [hunk 1]
```diff
@@ -56,6 +56,40 @@ void MolDraw2DCairo::drawLine(const Point2D &cds1, const Point2D &cds2) {
   cairo_stroke(dp_cr);
 }
 
+void MolDraw2DCairo::drawWavyLine(const Point2D &cds1, const Point2D &cds2,
+                                  const DrawColour &col1,
+                                  const DrawColour &col2,
+                                  unsigned int nSegments, double vertOffset) {
+  PRECONDITION(dp_cr, "no draw context");
+
+  if (nSegments % 2)
```
Comment (bp-kelley) [inline Code/GraphMol/MolDraw2D/MolDraw2DCairo.cpp]: Precondition on nsegments?

File: Code/GraphMol/MolDraw2D/MolDraw2DCairo.cpp [hunk 2]
```diff
@@ -56,6 +56,40 @@ void MolDraw2DCairo::drawLine(const Point2D &cds1, const Point2D &cds2) {
...
+    ++nSegments;  // we're going to assume an even number of segments
+
+  Point2D perp = calcPerpendicular(cds1, cds2);
+  Point2D delta = (cds2 - cds1);
+  perp *= vertOffset;
+  delta /= nSegments;
```
Comment (bp-kelley) [inline Code/GraphMol/MolDraw2D/MolDraw2DCairo.cpp]: Check if delta is zero length

File: Code/GraphMol/MolDraw2D/MolDraw2D.cpp [hunk 3]
```diff
@@ -1153,18 +1156,20 @@ void MolDraw2D::drawRect(const Point2D &cds1, const Point2D &cds2) {
   drawPolygon(pts);
 }
 
+void MolDraw2D::drawWavyLine(const Point2D &cds1, const Point2D &cds2,
+                             const DrawColour &col1, const DrawColour &col2,
+                             unsigned int nSegments, double vertOffset) {
+  drawLine(cds1, cds2, col1, col2);
```
Comment (bp-kelley) [inline Code/GraphMol/MolDraw2D/MolDraw2D.cpp]: RD_UNUSED(nSegments); or comment out in definition

File: Code/GraphMol/MolDraw2D/MolDraw2DCairo.cpp (hunk 2 above)
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/MolDraw2DCairo.cpp]: The code is happy to draw bonds of zero length. Should be able to draw zero-length wavy lines too. (These are obviously not useful, but neither are they really errors. Checking for them everywhere in order to generate warnings would add a fair amount of overhead for warnings that are very likely to be ignored)

File: Code/GraphMol/MolDraw2D/MolDraw2DCairo.cpp (hunk 1 above)
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/MolDraw2DCairo.cpp]: Done, both here and in the SVG drawer.

File: Code/GraphMol/MolDraw2D/MolDraw2D.cpp (hunk 3 above)
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/MolDraw2D.cpp]: Done

File: Code/GraphMol/MolDraw2D/MolDraw2DCairo.cpp (hunk 2 above)
Comment (bp-kelley) [inline Code/GraphMol/MolDraw2D/MolDraw2DCairo.cpp]: Ok, I'll merge when I get off of mobile

**Comment by greglandrum** [comment]
Merging based on previous comment.

**Pull request merged**